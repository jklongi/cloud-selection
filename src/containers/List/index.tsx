type Item = object

type Props<I extends Item> = {
    items: I[]
    renderItem: (item: I) => JSX.Element | null
    Container: React.ElementType
}

const List = <I extends Item>({ items, renderItem, Container }: Props<I>) => (
    <Container>{items.map(renderItem)}</Container>
)

export default List
