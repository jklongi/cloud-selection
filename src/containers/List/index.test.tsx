import { mount } from 'enzyme'
import List from './index'

const items = [{ id: 1 }, { id: 2 }, { id: 3 }]

describe('<List />', () => {
    it('renders its container', () => {
        const wrapper = mount(
            <List Container={'p'} items={items} renderItem={(item) => <span key={item.id}>{item.id}</span>} />,
        )
        expect(wrapper.find('p')).toHaveLength(1)
    })

    it('renders nodes equal to items', () => {
        const wrapper = mount(
            <List Container={'p'} items={items} renderItem={(item) => <span key={item.id}>{item.id}</span>} />,
        )
        expect(wrapper.find('span')).toHaveLength(items.length)
    })

    it('renders its children', () => {
        const wrapper = mount(
            <List Container={'p'} items={items} renderItem={(item) => <span key={item.id}>{item.id}</span>} />,
        )
        expect(wrapper.text()).toEqual('123')
    })
})
