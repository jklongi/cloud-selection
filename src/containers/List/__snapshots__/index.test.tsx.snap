// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<List /> matches snapshot 1`] = `ReactWrapper {}`;
